crates/task-impls/src/quorum_proposal_recv.rs

```diff
@@ -0,0 +1,409 @@
+
+                        // This is for the case where we form a QC but have not yet seen the previous proposal ourselves
+                        let should_propose = self.quorum_membership.get_leader(new_view)
+                            == self.public_key
+                            && high_qc.view_number
+                                == self.current_proposal.clone().unwrap().view_number;
+                        let qc = high_qc.clone();
+                        if should_propose {
+                            debug!(
+                                "Attempting to publish proposal after voting; now in view: {}",
+                                *new_view
+                            );
+                            let pdd = {
+
+                            }
+                            broadcast_event(
+                                Arc::new(HotShotEvent::ProposeNow(
+                                    qc.view_number + 1,
+
+                                ))
+                            )
+                            // self.publish_proposal_if_able(qc.view_number + 1, &event_stream)
+                            //     .await;
+                        }
+                        if self.vote_if_able(&event_stream).await {
+                            self.current_proposal = None;
+                        }
+                    }
+                    warn!("Failed liveneess check; cannot find parent either\n High QC is {:?}  Proposal QC is {:?}  Locked view is {:?}", high_qc, proposal.data.clone(), locked_view);
+
+                    return;
+                };
+
+                self.spawned_tasks
+                    .entry(proposal.data.get_view_number())
+                    .or_default()
+                    .push(async_spawn(
+                        validate_proposal(
+                            proposal.clone(),
+                            parent_leaf,
+                            self.consensus.clone(),
+                            self.decided_upgrade_cert.clone(),
+                            self.quorum_membership.clone(),
+                            parent_state.clone(),
+                            view_leader_key,
+                            event_stream.clone(),
+                            sender,
+                            self.output_event_stream.clone(),
+                            self.storage.clone(),
+                        )
+                        .map(AnyhowTracing::err_as_debug),
+                    ));
+            }
+            _ => {}
+        }
+    }
+}
+
+impl<TYPES: NodeType, I: NodeImplementation<TYPES>> TaskState
+    for QuorumProposalRecvTaskState<TYPES, I>
+{
+    type Event = Arc<HotShotEvent<TYPES>>;
+    type Output = ();
+    fn filter(&self, event: &Arc<HotShotEvent<TYPES>>) -> bool {
+        !matches!(
+            event.as_ref(),
+            HotShotEvent::QuorumProposalRecv(..) | HotShotEvent::QuorumProposalValidated(..),
```

Comment:
Why is the `QuorumProposalValidated` event needed here?

Reply:
Mistake on my part, removing